### ISSUE TITLE
Generated code references enum schema when using outputSchemaToDefault = true

### ISSUE THREAD

**Issue by ldaley** [jOOQ/jOOQ#11972]
### Expected behavior

When using `outputSchemaToDefault = true`, there should be no references to the schema used for generation.

### Actual behavior

At least for checks and function default values involving enum values, the input schema is referenced.

### Steps to reproduce the problem

See https://github.com/ldaley/jooq-enum-qualified-reference-bug-repro

### Versions

- jOOQ: 3.14.11
- Java: 11
- Database (include vendor): Postgres 11
- OS: Mac OS
- JDBC Driver (include name if inofficial driver): Postgres 11

**Comment by lukaseder** [comment]
Thanks for your report. I agree, that setting should affect generated enums, though if I recall correctly, some things will stop working when the enums aren't qualified (enum arrays, perhaps?).

**Comment by tomdcc** [comment]
@lukaseder unless someone has been fiddling with the system catalogs, the array type for an enum will always be in the same schema as the enum itself. Regardless, someone using this setting presumably is taking responsibility for setting the search path to find everything that they need.

**Comment by lukaseder** [comment]
> unless someone has been fiddling with the system catalogs, the array type for an enum will always be in the same schema as the enum itself.

I meant that PG has some issues looking up `x[]` when `x` is not fully qualified or something like that.

**Comment by martinlau** [comment]
Not sure if this is the same issue, or should be raised as a separate issue, but 3.17.0 with Postgres columns which are arrays of enums introduces a problem with `outputSchemaToDefault: true`.

Generated insert/update SQL renders `"my_column" = cast(? as ""."my_enum"[])` -- and the empty schema there ("".) blows things up. Fortunately outputSchemaToDefault is cosmetic in my use case, so happy to remove it, but other cases may not be able to.

**Comment by lukaseder** [comment]
Thanks for your comment, @martinlau. I'm currently releasing 3.17.1, which should be ready by the end of the day. You can verify whether your issue is fixed already by:

- Building 3.17.1-SNAPSHOT from github
- Using the snapshot provided from here, if you're licensed: https://www.jooq.org/download/versions

**Comment by lukaseder** [comment]
I just realised, this is a different issue from https://github.com/jOOQ/jOOQ/issues/13723, which is the issue that has been fixed in 3.17.1

I will check if this issue here is really related, soon.

**Comment by martinlau** [comment]
Yep - 3.17.1 seems to resolve my version of the issue.

Apologies that I didn't spot #13723 - I would have watched that rather than going a bit off topic here.

**Comment by lukaseder** [comment]
No worries at all

**Comment by lukaseder** [comment]
> I will check if this issue here is really related, soon.

Nope, not related. The generated code contains plain SQL references to the schema, which was supposed to be removed by the `<outputSchemaToDefault/>` flag. But that is currently not possible, because we don't parse / process such SQL fragments yet. The relevant feature request is this: https://github.com/jOOQ/jOOQ/issues/7850

It will be a good reminder to check if this `<outputSchemaToDefault/>` flag will work correctly in those expressions, once they're parsed via #7850
